libpromises/evalfunction.c

```diff
@@ -9683,6 +9698,8 @@ const FnCallType CF_FNCALL_TYPES[] =
                   FNCALL_OPTION_NONE, FNCALL_CATEGORY_DATA, SYNTAX_STATUS_NORMAL),
     FnCallTypeNew("concat", CF_DATA_TYPE_STRING, CONCAT_ARGS, &FnCallConcat, "Concatenate all arguments into string",
                   FNCALL_OPTION_VARARG, FNCALL_CATEGORY_DATA, SYNTAX_STATUS_NORMAL),
+    FnCallTypeNew("trim", CF_DATA_TYPE_STRING, TRIM_ARGS, &FnCallTrim, "Trim string",
+                  FNCALL_OPTION_NONE, FNCALL_CATEGORY_DATA, SYNTAX_STATUS_NORMAL),
```

Comment:
I just realized that the JIRA ticket explicitly says that the function shall be `string_trim` instead :facepalm: 

I will wait to get other feedback in the PR to do the rework at once.

Reply:
@nickanderson @olehermanse ^

Reply:
I don't have a strong opinion, but I prefer `string_trim()`, since most of the other string functions follow this pattern.

Reply:
I changed it to `string_trim` following the "new" convention.